repose-aggregator/components/services/datastore-service/datastore-service-impl/src/test/scala/org/openrepose/nodeservice/distributed/servlet/DistributedDatastoreServletTest.scala

```diff
@@ -0,0 +1,97 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.nodeservice.distributed.servlet
+
+import java.net.InetAddress
+import java.util.Collections
+import javax.servlet.http.HttpServletResponse
+import javax.servlet.http.HttpServletResponse._
+
+import io.opentracing.mock.MockTracer
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.openrepose.core.services.datastore.distributed.ClusterConfiguration
+import org.openrepose.core.services.datastore.distributed.config._
+import org.openrepose.core.services.datastore.impl.distributed.CacheRequest.CACHE_URI_PATH
+import org.openrepose.core.services.datastore.{Datastore, DatastoreAccessControl, DatastoreService}
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
+import org.springframework.mock.web.{MockHttpServletRequest, MockHttpServletResponse}
+
+
+@RunWith(classOf[JUnitRunner])
+class DistributedDatastoreServletTest extends FunSpec with BeforeAndAfterEach with Matchers with MockitoSugar {
+
+  private val mockDatastoreService = mock[DatastoreService]
+  private val mockDatastore = mock[Datastore]
+  Mockito.when(mockDatastoreService.getDefaultDatastore).thenReturn(mockDatastore)
+  private val distributedDatastoreConfiguration = mock[DistributedDatastoreConfiguration]
+  val mockTracer = new MockTracer
+  val distributedDatastoreServlet = new DistributedDatastoreServlet(
+    mockDatastoreService,
+    mock[ClusterConfiguration],
+    new DatastoreAccessControl(Collections.emptyList[InetAddress], true),
+    distributedDatastoreConfiguration,
+    mockTracer
+  )
+
+  var servletRequest: MockHttpServletRequest = _
+  var servletResponse: HttpServletResponse = _
+
+
+  override def beforeEach(): Unit = {
+    mockTracer.reset()
+    servletRequest = new MockHttpServletRequest
+    servletRequest.setRequestURI(CACHE_URI_PATH)
+    servletRequest.setProtocol("1.1")
+    servletResponse = new MockHttpServletResponse
+  }
+
+  describe("Distributed Datastore calls without a Cache Key") {
+    val methodsResults = Map(
+      // @formatter:off
+      "GET"     -> SC_NOT_FOUND,
+      "HEAD"    -> SC_NOT_FOUND,   // Head calls Get
+      "PUT"     -> SC_BAD_REQUEST, // No data sent
+      "PATCH"   -> SC_BAD_REQUEST, // No data sent
+      "POST"    -> SC_METHOD_NOT_ALLOWED,
+      "TRACE"   -> SC_METHOD_NOT_ALLOWED,
+      "CUSTOM"  -> SC_NOT_IMPLEMENTED,
+      "BOGUS"   -> SC_NOT_IMPLEMENTED,
+      "OPTIONS" -> SC_OK) // This is handled by the parent HTTP Servlet
+      // @formatter:on
```

Comment:
Holy carp, how did I just learn about this magic?! I need more `@formatter` in my life.